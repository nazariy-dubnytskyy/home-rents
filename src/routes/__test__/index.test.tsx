import * as React from 'react';
import '@testing-library/jest-dom';
import '@testing-library/jest-dom/extend-expect';

import { render, RenderResult, fireEvent } from '../../utils/test-utils';
import Routes from '..';

let renderResult: RenderResult;

describe('Routes component', () => {
  beforeEach(() => {
    renderResult = render(<Routes />);
  });

  it('renders correctly', () => {
    const { container } = renderResult;

    expect(container.firstChild).toMatchSnapshot();
  });

  it('Home link goes to Home page', () => {
    const { getByText } = renderResult;
    const link = getByText('Home').closest('a');

    fireEvent.click(link);
    expect(document.body.textContent).toContain('Home page');
  })

  it('Register link goes to Register page', () => {
    const { getByText } = renderResult;
    const link = getByText('Register').closest('a');

    fireEvent.click(link);
    expect(document.body.textContent).toContain('RegisterTenant page');
  })

});