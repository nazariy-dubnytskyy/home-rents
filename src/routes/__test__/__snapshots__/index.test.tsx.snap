// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes component renders correctly 1`] = `
<nav
  class="sc-AxjAm ddzPpr"
>
  <ul
    class="sc-AxirZ fvpbAG"
  >
    <li
      class="sc-AxiKw gmGsig"
    >
      <a
        href="/"
      >
        Home
      </a>
    </li>
    <li
      class="sc-AxiKw gmGsig"
    >
      <a
        href="/register-tenant"
      >
        Register
      </a>
    </li>
  </ul>
</nav>
`;
